lib/rust/ensogl/core/src/display/shape/compound/rectangle.rs

```diff
@@ -34,8 +36,8 @@ mod shape {
             // canvas area. Thus, we need to recompute the new canvas size for the scaled shape.
             let canvas_clip_height_diff = &canvas_height * (clip.y() * 2.0);
             let canvas_clip_width_diff = &canvas_width * (clip.x() * 2.0);
-            let canvas_height = canvas_height + &canvas_clip_height_diff;
-            let canvas_width = canvas_width + &canvas_clip_width_diff;
+            let canvas_height = canvas_height + &canvas_clip_height_diff.abs();
+            let canvas_width = canvas_width + &canvas_clip_width_diff.abs();
```

Comment:
I was trying to keep the shader as optimized as possible. This implementation gives the same possibilities with slightly slower shader, thus I prefer the previous one.